### PR TITLE
MAINT: Replace remainig Travis CI jobs with GitHub workflows

### PR DESCRIPTION
**Pull request by schacht-certat**
This commit adds a GitHub workflow to run nosetests, both with only the
    basic requirements installed and with all the requirements installed.
    The steps to setup the host system are in the script `setup-full.sh` in
    the workflow scripts folder, so we don't have to clutter the workflow
    definition.
    The workflow also runs codecov.
    Another workflow added in this commit builds the documentation using
    sphinx.
    
    The commit also adds a `skip_ci` check for tests, that skips the check
    if it runs on a host that has the environment variable CI set to true.
    This commit enables this check in the
    intelmq/tests/bots/experts/national_cert_contact_certat/test_expert.py
    because that one failes with GitHub workflows.

### CODE REVIEW

File: .github/workflows/nosetests.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#Github Workflow to build Debian packages for intelmq-api
```
Comment (wagner-certat) [inline .github/workflows/nosetests.yml:1]: leftover from copy & paste?

File: .github/workflows/nosetests.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      run: sudo apt-get install redis -y
+
+    - name: Install test dependencies
+      run: pip install nose Cerberus pyyaml requests_mock coverage
+
+    - name: Install dependencies
+      run: pip install -e .
+
+    - name: Run tests
+      run: INTELMQ_SKIP_INTERNET=1 nosetests --with-coverage --cover-package=intelmq --cover-branches
```
Comment (wagner-certat) [inline .github/workflows/nosetests.yml:41]: I'd prefer if we run also the tests requiring internet connectivity

File: .github/workflows/nosetests.yml (hunk 1 above)
Comment (schacht-certat) [reply .github/workflows/nosetests.yml]: fixed

File: .github/workflows/nosetests.yml (hunk 2 above)
Comment (schacht-certat) [reply .github/workflows/nosetests.yml]: fixed

File: intelmq/tests/bots/experts/abusix/test_expert.py [hunk 3]
```diff
@@ -33,7 +33,6 @@
 
 
 @test.skip_internet()
-@test.skip_travis()
```
Comment (wagner-certat) [inline intelmq/tests/bots/experts/abusix/test_expert.py:36]: These tests failed regularly in the past and to not cause confusion for pull requestors for failings tests, we disabled the flaky tests in the past.

File: intelmq/tests/bots/experts/cymru_whois/test_expert.py [hunk 4]
```diff
@@ -68,7 +68,6 @@
 
 @test.skip_redis()
 @test.skip_internet()
-@test.skip_travis()
```
Comment (wagner-certat) [inline intelmq/tests/bots/experts/cymru_whois/test_expert.py:71]: These tests failed regularly in the past and to not cause confusion for pull requestors for failings tests, we disabled the flaky tests in the past.

**Review by wagner-certat — COMMENTED** [review]
Sometimes pip is called with and sometimes without `sudo`. Is this mix intentional? (Apparently it works)

File: .github/workflows/build-docs.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+#
+name: "Build the documentation with sphinx"
+on:
+  push:
+    branches: [develop, maintenance, master]
+  pull_request:
+    branches: [develop, maintenance]
+
+jobs:
+  nosetest:
```
[the remark this answers is not in the document]
Comment (schacht-certat) [reply .github/workflows/build-docs.yml]: As ususal :blush: fixed.

File: .github/workflows/scripts/setup-full.sh [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+# Install the dependencies of all the bots
+pip install wheel
+for file in intelmq/bots/*/*/REQUIREMENTS.txt; do
+	echo $file;
+	pip install -r $file;
+done
+
+# Install the test dependencies
+pip install Cerberus!=1.3 codecov pyyaml requests_mock
```
[the remark this answers is not in the document]
Comment (schacht-certat) [reply .github/workflows/scripts/setup-full.sh]: Ah, you're right- I've removed the lines

File: intelmq/tests/bots/experts/abusix/test_expert.py (hunk 3 above)
Comment (schacht-certat) [reply intelmq/tests/bots/experts/abusix/test_expert.py:36]: Fixed

File: intelmq/tests/bots/experts/cymru_whois/test_expert.py (hunk 4 above)
Comment (schacht-certat) [reply intelmq/tests/bots/experts/cymru_whois/test_expert.py:71]: Fixed

**Comment by schacht-certat** [comment]
> Sometimes pip is called with and sometimes without `sudo`. Is this mix intentional? (Apparently it works)

Thats how it was in `.travis.yml` - when I tried to run everything with sudo, some pip installs failed. I would leave it that way.

**Review by wagner-certat — APPROVED** [review]

**Pull request merged**